File: lv1-library-loan/csharp/LibraryLoan.Application/Members/MemberRepositoryErrors.cs
using LibraryLoan.Domain.Members;
using LibraryLoan.Results;

namespace LibraryLoan.Application.Members;

public static class MemberRepositoryErrors
{
    public static Error NotFound(MemberId memberId) => new(
        "Member.NotFound",
        $"会員(ID: {memberId})が見つかりません。");
}

File: lv1-library-loan/typescript/src/application/members/member-repository-errors.ts
import type { MemberId } from "../../domain/members/member-id.js";
import { ResultError } from "../../results/result-error.js";

export const MemberRepositoryErrors = {
  notFound: (memberId: MemberId): ResultError =>
    new ResultError("Member.NotFound", `会員(ID: ${memberId})が見つかりません。`),
};
